wordpress-mobile/WordPress-Android
# Issue/13326 trigger fix threats

Parent #13326

This PR triggers fix threats action on scan state screen and handles result from `FixThreatsUseCase`.
Child PR for `FixThreatsUseCase`: #13796

As fix threats API is asynchronous, polling for fix status is done in a separate PR: #13798 
[PR: #13798 also includes threat fixing flow (web vs mobile) video] 

**To test**

Prerequisite:

1. Make sure both Scan feature flag and MySiteImprovements flag are set to on in the App settings.
2. You have access to WP.com account with a site having scan capability and threats (e.g. https://pressable-jetpack-daily-scan.mystagingwebsite.com). Server credentials should be set on the site.
3. I've added a threat to the site. In case you run out of threats, re-install bad calendar plugin from [here](https://jetpackp2.files.wordpress.com/2020/05/calendar-1.zip) and re-run scan.

- Login to the app using above WP.com account and select the site on the My Site tab
- Click scan menu item on My Site
- Click "Fix `x` threats" button on scan state screen with threats
- Notice that
   - A confirmation dialog is displayed
   - Action buttons are disabled when fix threats action is triggered (when ok clicked on confirmation dialog)
   - (Action) Started message is displayed when action succeeds
   - Error message is displayed when action fails
   - Action buttons are enabled when action fails

(Tests included in `ScanViewModelTest`)

Notes:

1. Ignore all styling
2. Server credentials check is not handled
3. Message texts are kept close to web version but might not match exactly. Ensure that they'er relevant and we can update them during copy review.

Merge Instructions

- ~~Make sure child PR #13796 is merged to develop~~ ✅ 
- ~~Make sure #13768 is merged to develop~~ ✅ 
- ~~Remove the "Not Ready for Merge" label~~ ✅ 
- Merge this PR

PR submission checklist:
